### PR TITLE
fix(elements-core): toc links yield 404 when using hash router

### PR DESCRIPTION
**Pull request by P0lip**
Fixes the issue with toc navigation in #2323

### CODE REVIEW

File: packages/elements-core/src/components/Docs/Article/index.tsx [hunk 1]
```diff
@@ -2,16 +2,22 @@ import { IMarkdownViewerProps } from '@stoplight/markdown-viewer';
...
+import { RouterTypeContext } from '../../../context/RouterType';
 import { MarkdownViewer } from '../../MarkdownViewer';
 import { DocsComponentProps } from '..';
 
 type ArticleProps = DocsComponentProps<IMarkdownViewerProps['markdown']>;
 
 const ArticleComponent = React.memo<ArticleProps>(({ data }) => {
+  const { pathname } = useLocation();
+  const routerKind = React.useContext(RouterTypeContext);
+  const basePath = routerKind === 'hash' ? `#${pathname.split('#')[0]}` : '';
```
Comment (Amjcraft) [inline packages/elements-core/src/components/Docs/Article/index.tsx:16]:
```suggestion
  const basePath = routerKind === 'hash' ? `${hash}` : '';
```

File: packages/elements-core/src/components/Docs/Article/index.tsx [hunk 2]
```diff
@@ -2,16 +2,22 @@ import { IMarkdownViewerProps } from '@stoplight/markdown-viewer';
...
+import { useLocation } from 'react-router-dom';
 
+import { RouterTypeContext } from '../../../context/RouterType';
 import { MarkdownViewer } from '../../MarkdownViewer';
 import { DocsComponentProps } from '..';
 
 type ArticleProps = DocsComponentProps<IMarkdownViewerProps['markdown']>;
 
 const ArticleComponent = React.memo<ArticleProps>(({ data }) => {
+  const { pathname } = useLocation();
```
Comment (Amjcraft) [inline packages/elements-core/src/components/Docs/Article/index.tsx:14]: pathname dose not return hash I believe. react-router docs says useLocations returns a hash prop.
https://github.com/remix-run/history/blob/main/docs/api-reference.md#location

Didn't test this out on our version though.

```suggestion
  const { hash } = useLocation();
```

File: packages/elements-core/src/components/Docs/Article/index.tsx (hunk 1 above)
Comment (Amjcraft) [reply packages/elements-core/src/components/Docs/Article/index.tsx:16]: Also just noticed the `#` before the pathname. Was that a typo or am I missing somthing?

Comment (P0lip) [reply packages/elements-core/src/components/Docs/Article/index.tsx:16]: no, that's expected.
We use double hashes for hash router.

image

http://localhost:63342/elements/a.html#/cyrh2afdvwnpb-test#baz

the first part is the actual route, while the other hash is the anchor, hence the split.
We split to get the route and append the anchor.

Comment (Amjcraft) [reply packages/elements-core/src/components/Docs/Article/index.tsx:16]: I see. Thanks for the clarification

**Review by Amjcraft — APPROVED** [review]
LGTM

**Pull request merged**